tests/ui/const-generics/generic_const_exprs/ice-generics_of-no-entry-found-for-key-113017.rs

```diff
@@ -0,0 +1,20 @@
+// test for ICE "no entry found for key" in generics_of.rs #113017
+
+#![feature(generic_const_exprs)]
+#![allow(incomplete_features)]
+
+pub fn String<V>(elem)
+//~^ ERROR expected one of `:`, `@`, or `|`, found `)`
+where
+    V: 'a,
+    //~^ ERROR use of undeclared lifetime name `'a`
+    for<const N: usize = { || {}}> V: 'a,
+    //~^ ERROR use of undeclared lifetime name `'a`
+    //~^^ ERROR only lifetime parameters can be used in this context
+    //~^^^ ERROR defaults for generic parameters are not allowed in `for<...>` binders
+    for<C2: , R2, R3: > <&str as IntoIterator>::Item: 'static,
+    //~^ ERROR `&` without an explicit lifetime name cannot be used here
+    //~^^ ERROR only lifetime parameters can be used in this context
+{}
```

Comment:
I would expecct that this file is still not minimized for this ICE
and believe that most of the syntax issues don't contribute to the ICE and make it more likely this tests ends up being ineffective once we change the way we handle such errors when continuing compilation.

It would be nice to manually spend a few minutes trying to fix all unrelated errors while tracking whether the ICE still triggers.

Reply:
well what you call "syntax issues" I call "additional coverage" :smile:

Reply:
what I mean is, we want tests to also protect against future ICEs and minimizing everything to the absolute minimum just reduces the effectiveness in that regard